File: 000-Mathematics/006.lcm.js
// Naive Approach - Time Complexity: O(a*b - max(a,b))

function lcmN(a, b) {
  let res = Math.max(a, b);
  while (true) {
    if (res % a == 0 && res % b == 0) {
      break;
    }
    res++;
  }
  return res;
}

// console.log(lcmN(8, 20));

// Efficient Solution

function gcd(a, b) {
  if (b == 0) {
    return a;
  }
  return gcd(b, a % b);
}

function lcm(a, b) {
  return parseInt((a * b) / gcd(a, b));
}

console.log(lcm(8, 20));

File: Geeks-For-Geeks/000-Mathematics/006.lcm.py
#Naive Approach - Time Complexity: O(a*b - max(a,b))

def lcmN(a,b):
    res = max(a,b)
    while True:
        if res % a == 0 and res % b == 0:
            break
        res += 1
    
    return res

print(lcmN(4,6))

# Efficient Solution: 
# LCM * HCF = a * b

def gcd(a,b):
    if b == 0:
        return a
    return gcd(b, a % b)

def lcm(a,b):
    return (a * b) // gcd(a,b)

print(lcm(4,10))